### ISSUE TITLE
add_tag directive doesn't work

### ISSUE THREAD

**Issue by workasr** [logstash-plugins/logstash-filter-truncate#2]
The add_tag directive doesn't seem to work.  No events get this tag added whether they were truncated or not.
NOTE: I am assuming that the expected behavior is that this tag would **only** be added if the field was truncated

> truncate {
>     id => "truncate-filter-global-1"  
>     add_tag => [ "truncated_by_length_filter" ]  
>     length_bytes => 20000  
>     fields => ["body"]  
> }

**Comment by Grundlefleck** [comment]
Also running into this issue, from what I can tell, for the likes of add_tag to work `filter_matched` would need to be called from this filter, and it doesn't appear to be.

Sources:
 * [docs](https://www.elastic.co/guide/en/logstash/current/_how_to_write_a_logstash_filter_plugin.html#_literal_filter_literal_method) to say filter_matched should be called
* truncate's [filter](https://github.com/logstash-plugins/logstash-filter-truncate/blob/master/lib/logstash/filters/truncate.rb#L41) method that does not have the call

I don't have a workaround I can suggest for you at the moment I'm afraid.

**Comment by jordansissel** [comment]
Ahh, good catch. This is a bug.

**Comment by Grundlefleck** [comment]
I did eventually find a workaround; to use a custom ruby block to truncate the message, and add a field/tag that can be used for later processing:

```
    ruby {
      code => "
      message_limit = 10000
      
      original_message = event.get('message')
      message_length = original_message.length
      if message_length > message_limit
        event.set('[@fields][truncated_by_logstash]', true)
        event.set('[@fields][original_size]', message_length)
        event.set('@message', original_message[0..message_limit] + '...')
        event.remove('message')
        event.tag('_truncated')    
      end
      "
    }
```
Works for us, but I certainly don't claim it's best practice, or elegant :)

**Comment by jsvd** [comment]
this has been fixed by #3

**Comment by corey-cole** [comment]
I'm using v1.0.2 of this plugin and am seeing that 100% of my log entries are tagged with value from 'add_tag' regardless of length.  Is this the expected behavior?
